internal/services/apimanagement/api_management_resource.go

```diff
@@ -741,6 +750,66 @@ func resourceApiManagementServiceCreateUpdate(d *pluginsdk.ResourceData, meta in
 
 	d.SetId(id.ID())
 
+	// Remove sample products and APIs after creating
+	if d.IsNewResource() && d.Get("remove_samples_on_create").(bool) {
+		d.Set("remove_samples_on_create", true)
+
+		apis := make([]apimanagement.APIContract, 0)
+
+		for apisIter, err := apiClient.ListByService(ctx, id.ResourceGroup, id.ServiceName, "", nil, nil, "", nil); apisIter.NotDone(); err = apisIter.NextWithContext(ctx) {
+			if err != nil {
+				return fmt.Errorf("listing APIs after creation of %s: %+v", id, err)
+			}
+			if apisIter.Response().IsEmpty() {
+				break
+			}
+			if apisList := apisIter.Values(); apisList != nil {
+				apis = append(apis, apisList...)
+			}
+		}
+
+		for _, api := range apis {
+			if api.ID != nil {
+				apiId, err := parse.ApiID(*api.ID)
+				if err != nil {
+					return fmt.Errorf("parsing API ID: %+v", err)
+				}
+				log.Printf("[DEBUG] Deleting %s", apiId)
+				if resp, err := apiClient.Delete(ctx, apiId.ResourceGroup, apiId.ServiceName, apiId.Name, "", utils.Bool(true)); err != nil && !utils.ResponseWasNotFound(resp) {
+					return fmt.Errorf("deleting %s: %+v", apiId, err)
+				}
+			}
+		}
+
+		products := make([]apimanagement.ProductContract, 0)
+
+		for productsIter, err := productsClient.ListByService(ctx, id.ResourceGroup, id.ServiceName, "", nil, nil, nil, ""); productsIter.NotDone(); err = productsIter.NextWithContext(ctx) {
+			if err != nil {
+				return fmt.Errorf("listing products after creation of %s: %+v", id, err)
+			}
+			if productsIter.Response().IsEmpty() {
+				break
+			}
+			if productList := productsIter.Values(); products != nil {
+				products = append(products, productList...)
+			}
+		}
+
+		for _, product := range products {
+			if product.ID != nil {
+				productId, err := parse.ProductID(*product.ID)
```

Comment:
we should move this left, then parse this insensitively:

```suggestion
			if product.ID == nil {
				continue
			}
			productId, err := parse.ProductIDInsensitively(*product.ID)
```